hoomd/hpmc/external/user.py

```diff
@@ -89,12 +98,21 @@ class CPPExternalPotential(ExternalField):
     Attributes:
         code (str): The code of the body of the external field energy function.
             After running zero or more steps, this property cannot be modified.
+        param_array ((*N*, ) `numpy.ndarray` of ``float``): Numpy
+            array containing dynamically adjustable elements in the potential
+            energy function as defined by the user. After running zero or more
+            steps, the array cannot be set, although individual values can still
+            be changed.
 
     """
 
-    def __init__(self, code):
-        param_dict = ParameterDict(code=str)
-        self._param_dict = param_dict
+    def __init__(self, code, param_array):
```

Comment:
```suggestion
    def __init__(self, code, param_array=[]):
```
Maintain the existing API for backwards compatibility.